src/processor/operator/persistent/reader/parquet/column_reader.cpp

```diff
@@ -270,36 +296,51 @@ void ColumnReader::decompressInternal(kuzu_parquet::format::CompressionCodec::ty
     case CompressionCodec::UNCOMPRESSED:
         throw common::CopyException("Parquet data unexpectedly uncompressed");
     case CompressionCodec::GZIP: {
-        throw common::NotImplementedException("ColumnReader::decompressInternal");
+        MiniZStream s;
+        s.Decompress(
+            reinterpret_cast<const char*>(src), srcSize, reinterpret_cast<char*>(dst), dstSize);
+        break;
     }
     case CompressionCodec::SNAPPY: {
         {
             size_t uncompressed_size = 0;
             auto res = kuzu_snappy::GetUncompressedLength(
                 reinterpret_cast<const char*>(src), srcSize, &uncompressed_size);
+            // LCOV_EXCL_START
             if (!res) {
-                throw std::runtime_error("Snappy decompression failure");
+                throw common::RuntimeException{"Snappy decompression failure"};
```

Comment:
Better be more specific on error messages, I think we should at least say this happens inside parquet column reader. the reason is we might have similar error message for this when we read from compressed csv files.